kong/plugins/ai-prompt-guard/handler.lua

```diff
@@ -1,112 +1,145 @@
+        if type(v.role) ~= "string" then
+          return nil, bad_format_error
+        end
+        if v.role == "user" then
+          if type(v.content) ~= "string" then
+            return nil, bad_format_error
+          end
+          buf:put(v.content)
+        end
       end
-    end
 
-    user_prompt = buf:get()
-
-  elseif request.messages then
-    -- just take the trailing 'user' prompt
-    for _, v in ipairs(request.messages) do
-      if v.role == "user" then
-        user_prompt = v.content
+      user_prompt = buf:get()
+
+    elseif type(request.messages) == "table" then
+      -- just take the trailing 'user' prompt
+      for _, v in ipairs(request.messages) do
+        if type(v.role) ~= "string" then
+          return nil, bad_format_error
+        end
+        if v.role == "user" then
+          if type(v.content) ~= "string" then
+            return nil, bad_format_error
+          end
+          user_prompt = v.content
+        end
       end
-    end
 
-  elseif request.prompt then
-    user_prompt = request.prompt
+    elseif type(request.prompt) == "string" then
+      user_prompt = request.prompt
 
-  else
-    return nil, "ai-prompt-guard only supports llm/v1/chat or llm/v1/completions prompts"
-  end
+    else
+      return nil, bad_format_error
+    end
+
+    if not user_prompt then
+      return nil, "no 'prompt' or 'messages' received"
+    end
 
-  if not user_prompt then
-    return nil, "no 'prompt' or 'messages' received"
-  end
 
-  -- check the prompt for explcit ban patterns
-  if conf.deny_patterns and #conf.deny_patterns > 0 then
-    for _, v in ipairs(conf.deny_patterns) do
+    -- check the prompt for explcit ban patterns
+    for _, v in ipairs(conf.deny_patterns or EMPTY) do
       -- check each denylist; if prompt matches it, deny immediately
       local m, _, err = ngx_re_find(user_prompt, v, "jo")
       if err then
-        return nil, "bad regex execution for: " .. v
+        -- regex failed, that's an error by the administrator
+        kong.log.err("bad regex pattern '", v ,"', failed to execute: ", err)
+        return kong.response.exit(500)
 
       elseif m then
         return nil, "prompt pattern is blocked"
       end
     end
-  end
 
-  -- if any allow_patterns specified, make sure the prompt matches one of them
-  if conf.allow_patterns and #conf.allow_patterns > 0 then
-    local valid = false
 
-    for _, v in ipairs(conf.allow_patterns) do
+    if #(conf.allow_patterns or EMPTY) == 0 then
+      -- no allow_patterns, so we're good
+      return true
+    end
+
+    -- if any allow_patterns specified, make sure the prompt matches one of them
+    for _, v in ipairs(conf.allow_patterns or EMPTY) do
       -- check each denylist; if prompt matches it, deny immediately
       local m, _, err = ngx_re_find(user_prompt, v, "jo")
 
       if err then
-        return nil, "bad regex execution for: " .. v
+        -- regex failed, that's an error by the administrator
```

Comment:
Same as above: it seems it would be more efficient to check validity of the regexes when the schema is created, instead of when it is being used.